### PR TITLE
ENYO-5212: Prevent Spottable Disabled OnClick

### PR DESCRIPTION
**Pull request by teckliew**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Check for `disabled` and `isHovered`

### Links
[//]: # (Related issues, references)
ENYO-5212

### Comments
Enact-DCO-1.0-Signed-off-by: Teck Liew teck.liew@lge.com

### CODE REVIEW

**Comment by teckliew** [comment]
I thought it would be a universal case to cover for `disabled` and `isHovered`, but I agree it'll be safer to limit the scope of this fix to checking for `disabled` in `SliderBehaviorDecorator`.

**Comment by viodragon2** [comment]
I just realized this problem can happen to all `disabled` components. When a disabled component rerenders, then `Spottable`'s internal state `isFocused` will be updated to `true` which in turn will add spottable class and hightlight at render time. I wonder if it's better to stop propagation in `onFocus` from `Spottable`?

**Comment by ryanjduffy** [comment]
Good analysis, @viodragon2! Appears that the exposure is limited to components that handle `onFocus` to cause a state change and therefore a prop change to Spottable triggering its state change. The new slider is, of course, one of those instances: SliderBehaviorDecorator handles `onFocus` to pass the focused state down to show the tooltip.

File: packages/spotlight/Spottable/Spottable.js [hunk 1]
```diff
@@ -316,14 +316,16 @@ const Spottable = hoc(defaultConfig, (config, Wrapped) => {
 		}
 
 		handleFocus = (ev) => {
-			if (ev.currentTarget === ev.target) {
-				this.isFocused = true;
-			}
+			if (!this.props.disabled) {
```
Comment (ryanjduffy) [inline packages/spotlight/Spottable/Spottable.js]: Let's update this to be an early bail out to minimize the diff and keep the code as readable as possible.

```js
handleFocus = (ev) => {
  if (this.props.disabled) return;
  ...
}
```

**Pull request merged**